bot/src/discord/schedule_helper.js

```diff
@@ -0,0 +1,306 @@
+const dbConnection = require('kotoba-node-common').database.connection;
+const WordScheduleModel = require('kotoba-node-common').models.createWordSchedulesModel(dbConnection);
+const showRandomWord = require('./show_random_word.js');
+
+const frequencyCheck = 60 * 1000;
+
+const statusConstants = {
+  running: 'running',
+  paused: 'paused',
+};
+
+/**
+ * Starts the interval
+ * @param {Monochrome} monochrome
+ * @param {boolean} firstCall
+ */
+async function setTimer(monochrome, firstCall) {
+  let now = (new Date()).getTime();
+  // The first time this function is called we don't want to run the command,
+  // just set the timer so it runs in sync with the frequencyCheck value
+  if (!firstCall) {
+    const pendingSchedules = await WordScheduleModel.find({ status: statusConstants.running });
```

Comment:
Would be great if we could query the database for only the ones that are due to run now, since in theory we could wind up with thousands of schedules in the database. But if that was sticky we can leave this be and maybe I'll take a look at it later.

Reply:
Would something like this work?

```js
    const pendingSchedules = await WordScheduleModel.find({
      status: statusConstants.running,
      start: { $lte: now },
      $or: [ { lastSent: null }, { $expr: { $lte: [ "$frequency" , { $subtract: [now, "$lastSent"] } ] } } ]
    });
```
Here's an example:
https://mongoplayground.net/p/WiEe_hN-ZzW

Reply:
Yeah, nice, that looks like it would work. I see, I forgot the null check in my example.

Reply:
What example? Did I miss something? With this approach, should we store lastSent as Number instead of Date? I don't know if the operators and comparers work fine with Date objects, also frequency and start are stored as Number. Would you prefer that they're all stored as Date instead?

Reply:
I meant the code I wrote at the bottom of this comment: https://github.com/mistval/kotoba/pull/148#discussion_r528034721

lastSent and start could both be dates, and frequency could be a number in ms. The Mongo operators should work fine (subtracting a number X from a date Y returns a new date that is X ms before Y). Or they could all be numbers, that would be fine too. Either way really. I don't think frequency makes sense as a date though, since it's an interval.

Reply:
The query wasn't working, it seems that the $subtract returns a number instead a Date, so I had to conver it to Date again:

```js
      const dueSchedules = await WordScheduleModel.find({
        status: statusConstants.running,
        start: { $lte: now },
        $or: [{ lastSent: null }, { $expr: { $lt: ['$lastSent', { $toDate: { $subtract: [now, '$frequency'] } }] } }],
      });
```

Reply:
This is fine. I'm wondering what `now` is though. Whether it's a Number or a Date object. If it's Date (`new Date()`) then I think it shouldn't need extra conversion.